apps/balancer-tools/src/lib/dune.ts

```diff
@@ -44,4 +51,28 @@ export class DuneAPI {
       )
       .catch((error: { message: string }) => ({ error: error.message }));
   }
+
+  async getTVLByRoundId(
+    roundId: number = 0,
+  ): Promise<undefined | TVLData[] | ErrorReturn> {
+    invariant(roundId > 0, "An valid roundId must be passed");
+    const roundData = Round.getRoundByNumber(roundId);
+    // Ex for Round 1:
+    // https://dune.com/queries/2842639?2.+Start+date_d83555=2022-04-13+00%3A00%3A00
+    const queryId = 2842639;
+    const parameters = [
+      QueryParameter.number(
+        "End date",
+        roundData.endDate.toISOString().slice(0, 10),
+      ),
+    ];
+
+    return this.client
+      .refresh(queryId, parameters)
```

Comment:
#question do we really need to call refresh? Couldn't we use a cached version of that data from a result that's already run? I wonder if that would make us spend less credits.

Reply:
This should be cacheable, I have to look into how Next does it.

I think the a `import { cache } from 'react'` should sufice, but not 100% sure on that